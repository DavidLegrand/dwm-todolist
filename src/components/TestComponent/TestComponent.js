import React, { useState, useEffect } from "react";
import ThemeSwitcher from "components/ThemeSwitcher";

const TestComponent = () => {
  const initialState = {
    firstName: "David",
    lastName: "Legrand",
    age: 32
  }

  const [user, setuser] = useState(initialState)
  const [posts, setPosts] = useState(null)

  useEffect(() => console.log("Effect : A chaque render de mon composant"))
  useEffect(() => console.log("Effect : A chaque modification de user"), [user])
  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch("https://jsonplaceholder.typicode.com/posts")
      setPosts(await response.json())
    }
    fetchData()
  }, [])

  const handleClick = () => setuser((previousUser) => ({ ...previousUser, age: previousUser.age + 1 }))
  
  return <>
    <h1>User {user.firstName} {user.lastName}, is {user.age} </h1>
    <button onClick={handleClick}>Happy Birthday !</button>
    <ThemeSwitcher />
    <ul>
      {posts && posts.map((post) => <li key={post.id}>{post.title}</li>)}
    </ul>
  </>;
};

export default TestComponent;
