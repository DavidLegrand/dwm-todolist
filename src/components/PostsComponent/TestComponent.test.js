import React from "react";
import { shallow } from "enzyme";
import TestComponent from "./PostsComponent";

describe("TestComponent", () => {
  test("matches snapshot", () => {
    const wrapper = shallow(<TestComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(2 + 2).toBe(4)
  });
});