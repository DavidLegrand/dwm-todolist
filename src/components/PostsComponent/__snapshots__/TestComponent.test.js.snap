// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent matches snapshot 1`] = `ShallowWrapper {}`;
